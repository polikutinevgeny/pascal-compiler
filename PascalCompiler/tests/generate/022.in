begin
    write(integer(5.5));
    write(real(3));
    write(char(33));
    write(integer('!'));
end.