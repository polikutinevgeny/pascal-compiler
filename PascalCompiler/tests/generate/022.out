.686p
include \masm32\include\masm32rt.inc
.mmx
.xmm
.code
main PROC
Push Ebp
Mov Ebp, Esp
Sub Esp, 0
Push 040160000h
Push 000000000h
Cvttsd2si Eax, qword ptr [Esp]
Add Esp, 4
Mov dword ptr [Esp], Eax
Mov Eax, Esp
printf("%d", dword ptr [Eax])
Sub Esp, 4
Push 3
Cvtsi2sd Xmm0, dword ptr [Esp]
Sub Esp, 4
Movsd qword ptr [Esp], Xmm0
Mov Eax, Esp
printf("%f", qword ptr [Eax])
Sub Esp, 8
Push 33
Mov Eax, Esp
printf("%c", dword ptr [Eax])
Sub Esp, 4
Push 33
Mov Eax, Esp
printf("%d", dword ptr [Eax])
Sub Esp, 4
Mov Esp, Ebp
Pop Ebp
exit
main ENDP
END main
